src/gpodder/gtkui/main.py

```diff
@@ -1963,125 +1978,57 @@ def treeview_available_show_context_menu(self, treeview, event=None):
-                share_menu = self._add_sub_menu(menu, _('Send to'))
-
-                item = Gtk.ImageMenuItem(_('Local folder'))
-                item.set_image(Gtk.Image.new_from_icon_name('folder', Gtk.IconSize.MENU))
-                self._submenu_item_activate_hack(item, self.save_episodes_as_file, episodes)
-                share_menu.append(item)
-                if self.bluetooth_available:
-                    item = Gtk.ImageMenuItem(_('Bluetooth device'))
-                    item.set_image(Gtk.Image.new_from_icon_name('bluetooth', Gtk.IconSize.MENU))
-                    self._submenu_item_activate_hack(item, self.copy_episodes_bluetooth, episodes)
-                    share_menu.append(item)
+            # Extensions section
+            entries = [(label, None if func is None else lambda a, b: func(episodes))
+                for label, func in list(gpodder.user_extensions.on_episodes_context_menu(episodes) or [])]
+            self.episode_context_menu_helper.replace_entries(entries)
 
-            menu.append(Gtk.SeparatorMenuItem())
-
-            item = Gtk.CheckMenuItem(_('New'))
-            item.set_active(any_new)
-            if any_new:
-                item.connect('activate', lambda w: self.mark_selected_episodes_old())
+            # 'Send to' submenu
+            if downloaded:
+                if self.sendto_menu.get_n_items() < 1:
+                    self.sendto_menu.insert_submenu(
+                        0, _('Send to'),
+                        self.application.builder.get_object('episodes-context-sendto'))
             else:
-                item.connect('activate', lambda w: self.mark_selected_episodes_new())
-            menu.append(item)
-
-            if can_lock:
-                item = Gtk.CheckMenuItem(_('Archive'))
-                item.set_active(any_locked)
-                item.connect('activate',
-                             lambda w: self.on_item_toggle_lock_activate(
-                                 w, False, not any_locked))
-                menu.append(item)
+                self.sendto_menu.remove_all()
 
-            menu.append(Gtk.SeparatorMenuItem())
-            # Single item, add episode information menu item
-            item = Gtk.ImageMenuItem(_('Episode details'))
-            item.set_image(Gtk.Image.new_from_icon_name('dialog-information',
-                                                        Gtk.IconSize.MENU))
-            item.set_action_name('win.toggleShownotes')
-            menu.append(item)
-
-            if len(self.get_selected_episodes()) == 1:
-                item = Gtk.MenuItem(_('Open download folder'))
-                item.connect('activate', self.on_open_episode_download_folder)
-                menu.append(item)
+            # New and Archive state
+            self.gPodder.lookup_action('episodeNew').change_state(GLib.Variant.new_boolean(any_new))
+            self.gPodder.lookup_action('episodeLock').change_state(GLib.Variant.new_boolean(any_locked))
```

Comment:
This should be `can_lock and any_locked` so the item is disabled for deleted episodes.

Reply:
Ignore the any_locked part, it should be disabled if not can_locked. set_episode_actions() handles disabling for actions shared between menus but this uses a different action.

Reply:
Should these actions be added to `set_episodes_actions()` then?

Reply:
It only applies to the context menu but the old and new code have no other set_enabled calls, so sure, add it to set_episode_actions() below an `# episode context menu` comment. I've always wanted to merge the two episode menus for consistency, and I think this new code will make that easier. Right after we figure out which menu has the better new/lock behavior. :)